sql/connect/server/src/main/scala/org/apache/spark/sql/connect/pipelines/PipelinesHandler.scala

```diff
@@ -129,6 +131,46 @@ private[connect] object PipelinesHandler extends Logging {
     }
   }
 
+  /**
+   * Block unsupported SQL commands that are not explicitly allowlisted.
+   */
+  def blockUnsupportedSqlCommand(queryPlan: LogicalPlan): Unit = {
+    val supportedCommand = Set(
+      classOf[DescribeRelation],
+      classOf[ShowTables],
+      classOf[ShowTableProperties],
+      classOf[ShowNamespacesCommand],
+      classOf[ShowColumns],
+      classOf[ShowFunctions],
+      classOf[ShowViews],
+      classOf[ShowCatalogsCommand],
+      classOf[ShowCreateTable])
+    val isSqlCommandExplicitlyAllowlisted = {
+      supportedCommand.exists(c => queryPlan.getClass.getName.equals(c.getName))
+    }
+    val isUnsupportedSqlPlan = if (isSqlCommandExplicitlyAllowlisted) {
+      false
+    } else {
+      // If the SQL command is not explicitly allowlisted, check whether it belongs to
```

Comment:
It took me a minute to parse what was going on here. Is my understanding correct that we basically disallow all query plans that are descendants of the plan classes here, **except** for some subclasses of `Command` that we know have no side effects?